### ISSUE TITLE
clean up Greaselion temp folders for existing profiles

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#24067]
## Description 

Found while testing https://github.com/brave/brave-browser/issues/21866

Fix for https://github.com/brave/brave-browser/issues/21866 works for new profiles, but not for existing profiles.

Older profiles can have thousands of temp folders which don't get cleaned up.

## Steps to Reproduce

   1. Have an old profile
   2. Look at `<Profile dir>/Greaselion/Temp` 
   3. See many, many greaselion folders

## Actual result:

they don't get cleaned up:

Example | Example
-- | --
<img width="262" alt="1" src="https://user-images.githubusercontent.com/28145373/179070519-0d7f81f4-2c84-4a50-99a2-395b437377ab.png"> | <img width="889" alt="2" src="https://user-images.githubusercontent.com/28145373/179070558-f548eb81-c4d9-4321-863a-e6fe104b4563.png">

## Expected result:

Expected the old, unused/un-needed folders to get cleaned up as they do on new profiles

## Reproduces how often: 

easily

## Brave version (brave://version info)

Brave | 1.42.63 Chromium: 103.0.5060.114 (Official Build) dev (x86_64)
-- | --
Revision | a1c2360c5b02a6d4d6ab33796ad8a268a6128226-refs/branch-heads/5060@{#1124}
OS | macOS Version 12.4 (Build 21F79)

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @zenparsing

**Comment by CptAmericaVsMonsters** [comment]
so... Can the folders be deleted?

**Comment by bsclifton** [comment]
@CptAmericaVsMonsters yes, you can delete the folders. If anything is needed, it should recreate it

**Comment by CptAmericaVsMonsters** [comment]
thanks

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
